libs/langchain-community/src/vectorstores/zep_cloud.ts

```diff
@@ -0,0 +1,347 @@
+): Record<string, unknown> | undefined {
+  if (typeof value === "object" && value !== null) {
+    return value as Record<string, unknown>;
+  }
+  if (value !== undefined) {
+    console.warn("Metadata filters must be an object, Record, or undefined.");
+  }
+  return undefined;
+}
+
+/**
+ * Interface for the configuration options for a ZepCloudVectorStore instance.
+ */
+export interface IZepCloudConfig {
+  apiKey?: string;
+  client?: ZepClient;
+  collectionName: string;
+  description?: string;
+  metadata?: Record<string, never>;
+}
+
+/**
+ * Interface for the parameters required to delete documents from a
+ * ZepCloudVectorStore instance.
+ */
+export interface IZepCloudDeleteParams {
+  uuids: string[];
+}
+
+/**
+ * ZepCloudVectorStore is a VectorStore implementation
+ * that uses the Zep long-term memory store as a backend.
+ *
+ * If the collection does not exist, it will be created automatically.
+ *
+ * Requires `@getzep/zep-cloud` to be installed:
+ *
+ *
+ * @property {ZepClient} client - The ZepClient instance used to interact with Zep's API.
+ * @property {Promise<void>} initPromise - A promise that resolves
+ * when the collection is initialized.
+ */
+export class ZepCloudVectorStore extends VectorStore {
+  public client: ZepClient;
+
+  public collectionName: string;
+
+  private readonly initPromise: Promise<void>;
+
+  constructor(embeddings: EmbeddingsInterface, args: IZepCloudConfig) {
+    super(embeddings, args);
+
+    this.initPromise = this.initCollection(args).catch((err) => {
+      console.error("Error initializing collection:", err);
+      throw err;
+    });
+  }
+
+  /**
+   * Initializes the document collection. If the collection does not exist, it creates a new one.
+   *
+   * @param {IZepConfig} args - The configuration object for the Zep API.
+   */
+  private async initCollection(args: IZepCloudConfig) {
+    if (args.client) {
+      this.client = args.client;
+    } else {
+      this.client = new ZepClient({
+        apiKey: args.apiKey,
+      });
+    }
+    try {
+      this.collectionName = args.collectionName;
+      await this.client.document.getCollection(this.collectionName);
+    } catch (err) {
+      // eslint-disable-next-line no-instanceof/no-instanceof
+      if (err instanceof Error) {
+        // eslint-disable-next-line no-instanceof/no-instanceof
+        if (err instanceof NotFoundError || err.name === "NotFoundError") {
+          await this.createCollection(args);
+        } else {
+          throw err;
+        }
+      }
+    }
+  }
+
+  /**
+   * Creates a new document collection.
+   *
+   * @param {IZepConfig} args - The configuration object for the Zep API.
+   */
+  private async createCollection(args: IZepCloudConfig) {
+    await this.client.document.addCollection(args.collectionName, {
+      description: args.description,
+      metadata: args.metadata,
+    });
+  }
+
+  /**
+   * @deprecated Adding vectors in unsupported in Zep Cloud
```

Comment:
Deprecated is the wrong framing - just throw an error

Reply:
Thanks, addressed